### PR TITLE
feat(medusa): add currency to region responses

### PR DESCRIPTION
**Pull request by riqwan**
Resolves https://github.com/medusajs/medusa/issues/5837

### CODE REVIEW

File: integration-tests/api/package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "license": "MIT",
   "private": true,
   "scripts": {
-    "test:integration": "jest --silent=false --maxWorkers=50% --bail --detectOpenHandles --forceExit --logHeapUsage",
+    "test:integration": "jest -u --silent=false --maxWorkers=50% --bail --detectOpenHandles --forceExit --logHeapUsage",
```
Comment (pKorsholm) [inline integration-tests/api/package.json]: question: Is there a reason to automatically update snapshots?

I don't think we should always update snapshots, I think having it be a manual process to see that a snapshot has changed and re-running tests w/ `-u` is a preferable to catch any potential changes.

**Review by pKorsholm — APPROVED** [review]
lgtm, just have a comment in the integration tests

File: integration-tests/api/package.json (hunk 1 above)
Comment (riqwan) [reply integration-tests/api/package.json]: good catch, yes i added it to update snapshots but forgot to remove it.

**Review by olivermrbl — APPROVED** [review]
LGTM, very nice clean up too!

**Pull request merged**